### ISSUE TITLE
[Feat]: Improve pod startup time

### ISSUE THREAD

**Issue by tonygilkerson** [project-zot/zot#2413]
### Is your feature request related to a problem? Please describe.

Our zot pod takes over 3min to start and we have a very small data set, the registry volume is ~8M and the s3 bucket is ~1.3G.  The readinessProbe seems to be waiting on "parsing storage and initializing" where it inspect each repo.  I would expect the pod to become ready in under 30 seconds. As it is now this will not scale as we add more data.  I have included our zot-config below. Does anyone have some advice for how I can speed up my startup?

### Describe the solution you'd like

When the Zot pod is restarted it would be nice to have a flag to turn off the consistency check between the s3 bucket and the metadb on the PV

### Describe alternatives you've considered

I have not found an alternative.  For now I have extended my liveness and readness probe to 5 minutes

**Comment by rchincha** [comment]
Maybe there is a "fastRestart": true flag that can be added so that we assume users know what they doing and zot can be less conservative.

**Comment by ktarplee** [comment]
Someone could theoretically change the S3 bucket at any time to make it out of sync with the local DB.  Why only "re-sync" at startup.  Gremlins don't only come out when Zot is turned off.  Maybe re-sync should be a periodic activity (but don't block startup on the re-sync).

**Comment by andaaron** [comment]
Well yes, but the case we would expect to be more common is for someone to use an older version of zot, and when upgrading and starting the new zot the data to be in a different format which could cause some errors if we don't handle it correctly.

In practice right now this could happen only if someone uses an older RC release and switches to 2.0+.
We don't support patching that old data in the DB for the RC releases, so we read the storage and make sure we add the entries in the format we need when zot starts.

**Comment by ktarplee** [comment]
Why not include the version of zot in the database.  When it different from the zot binary version (forward or backward) start the sync process.  If it is the same then bypass the sync process.

**Comment by ktarplee** [comment]
The real issue is that the startup time for ZOT is a function of the amount of data in S3.  That is a bad property.  Even databases do not have that property (they keep there data on disk and only load limited metadata into memory and do no block startup on that process).

A solution to this is to start ZOT but in a degraded state if the sync has not completed yet.   The sync process can complete in the background and once completed Zot is fully functional.

**Comment by USA-RedDragon** [comment]
Seeing if talking in this issue will reopen it 🤞 

Have a PR proposal to continue the discussion of this issue: https://github.com/project-zot/zot/pull/4026

**Comment by andaaron** [comment]
Can we now mark this as resolved, since https://github.com/project-zot/zot/pull/4026 is merged?
